src/helm/benchmark/presentation/summarize.py

```diff
@@ -759,16 +760,27 @@ def create_group_tables_by_metric_group(self, group: RunGroup) -> List[Table]:
         Each table has `adapter_spec`s as rows and scenarios or groups as columns."""
         tables: List[Table] = []
         adapter_to_runs: Dict[AdapterSpec, List[Run]] = defaultdict(list)
-        all_metric_groups: List[str] = []
         subgroups = self.expand_subgroups(group)
         for subgroup in subgroups:
-            all_metric_groups.extend(subgroup.metric_groups)
             for adapter_spec, runs in self.group_adapter_to_runs[subgroup.name].items():
                 coarse_adapter_spec = get_coarse_adapter_spec(adapter_spec, adapter_keys_shown=group.adapter_keys_shown)
                 filtered_runs = self.filter_runs_by_visibility(runs, group)
                 if filtered_runs:
                     adapter_to_runs[coarse_adapter_spec].extend(filtered_runs)
-        all_metric_groups = list(dict.fromkeys(all_metric_groups))  # deduplicate while preserving order
+
+        all_metric_groups: List[str] = []
+        if group.metric_groups:  # if the group specifies the metric groups, use them
+            all_metric_groups = group.metric_groups
+        else:  # otherwise, collect them from subgroups
+            for subgroup in subgroups:
+                all_metric_groups.extend(subgroup.metric_groups)
+            # deduplicate, remove basic metric group if we include the detailed one, remove hidden metric groups
+            all_metric_groups = [
+                metric_group
+                for metric_group in dict.fromkeys(all_metric_groups)
```

Comment:
Just do `set(all_metric_groups)` instead of `dict.fromkeys(all_metric_groups)`

Reply:
Oh, I'm using `dict` to preserve the order of the metric_groups. (For `set` the order is undefined)